frontend/src/pages/_app.page.tsx

```diff
@@ -65,11 +66,19 @@ function MyApp({ Component, pageProps }: AppProps) {
       titleCase: false,
       debug: process.env.NEXT_PUBLIC_DEBUG === "true",
     });
-    ReactGa.set({
+    const gaFields: ReactGa.FieldsObject = {
       anonymizeIp: true,
       transport: "beacon",
-    });
+    };
     const cookies = document.cookie.split("; ");
+    cookies.forEach((item) => {
+      if (item.trim().startsWith("utm_")) {
+        const cookieId = item.split("=")[0];
+        Object.assign(gaFields, convertUtmCookieToGaField(item));
+        clearCookie(cookieId);
```

Comment:
(question, blocking) I think this code might be clearing the cookie early when it first finds a `utm_` value even though there are other keys that start with `utm_`. 
(comment) Glad we are clearing the data after we use them for analytics!

Reply:
It shouldn't be. I tested and saw all the GA fields ARE assigned. It only clears the single cookie matching the `cookieId` value of the `item` that it just assigned to the `gaFields`.